src/fsutil.rs

```diff
@@ -0,0 +1,17 @@
+use std::{fs, io, path::Path};
+
+pub(crate) fn copy_dir_all(src: impl AsRef<Path>, dst: impl AsRef<Path>) -> io::Result<()> {
+    fs::create_dir_all(&dst)?;
+    for entry in fs::read_dir(src)? {
+        let entry = entry?;
+        let ty = entry.file_type()?;
+        if ty.is_dir() {
+            if entry.file_name() != ".git" {
```

Comment:
In fact that excludes the `.git` directory, but what if there are `.svn` or other things inside. I believe I need to land #245 soon, so that we don't have to exclude anything at all.

Reply:
Sure, but allowing `.git` has a serious risc of trashing an existing repo into which we're generating the template.

Haven't tried this, but using `--git`, which initializes the destination as a new repository, aren't the `.git` folder in the actual template also ignored there? I believe it is.

Reply:
Sure enough, generating from a git repository runs `remove_history()` on the cloned repository - this in fact just deletes the `.git` folder too.

Reply:
@sassman Given my latest push goes through on this, I've made `--git` and `--dir` mutually exclusive!
I've also introduced `--pick` in order to clear up the functionality (and pave the way for #291)